### ISSUE TITLE
label leakage

### ISSUE THREAD

**Issue by Tianbaojie** [DaDaCheng/WalkPooling#4]
https://github.com/DaDaCheng/WalkPooling/blob/df01cfc293e429d09ad51f83bc831f3ca9ff098c/src/main.py#L182

hi, your code is causing label leakage.
You used the test and validation sets and edge information during training

**Comment by DaDaCheng** [comment]
No, in _utils/split_edges_, we separate the training, validation, and test edges, 

and for example in _utils/set_init_attribute_representation_, if you set `observe_val_and_injection==False`, the observed graph is only constructed by training edges; if you set `observe_val_and_injection==True`, the observed graph is constructed by training and validation edges, but not test edges.

```
def set_init_attribute_representation(data,args):
    #Construct data_observed and compute its node attributes & representation
    edge_index = torch.cat((data.train_pos,data.train_pos[[1,0],:]),dim=1)
    if args.observe_val_and_injection == False:
        data_observed = Data(edge_index=edge_index)
    else:
        edge_index=torch.cat((edge_index,data.val_pos,data.val_pos[[1,0],:]),dim=1)
        data_observed = Data(edge_index=edge_index)
```
Thanks for your interest in our work, but please read our paper and code in detail before making such claims.
